### PR TITLE
Convert site from CRA to Gatsby + NetlifyCMS

### PR DESCRIPTION
**Pull request by ironng**
This encompasses the first move from the Create React App environment to Gatsby and NetlifyCMS. There's nothing CMS'd yet, but the website looks the same!

This also updates the Bootstrap version from 3.3.7 to 4.4.1 and swaps out React Bootstrap for Reactstrap, and makes the requisite style adjustments as a result of this change. All less files are converted to scss files. Unused components have also been removed.

There is also a change to the circleci config to help it be less broken, but I'm guessing this will be removed entirely for a github workflow.

Redux/api calls will be hooked up in a separate PR, as well as converting pages to be editable via CMS, etc.

### CODE REVIEW

**Comment by ironng** [comment]
@Shadowfiend I created the branch `old-sport` to be the base branch for our switch over to Gatsby land (Can you see the green light in the distance?). This PR is just the start...I figured it's better if we do this in parts. :)

Also, any thoughts on fixing the circleci builds, or can we just wait to switch over to the github workflow?

**Comment by Shadowfiend** [comment]
We can actually add the GitHub workflow here directly, and disable the Circle one when we merge everything back.

**Comment by Shadowfiend** [comment]
Let me push some actions to this branch… Need to do a light infra touch first.

File: gatsby-config.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    {
+      resolve: 'gatsby-plugin-purgecss', // purges all unused/unreferenced css rules
+      options: {
+        develop: true, // Activates purging in npm run develop
+        purgeOnly: ['/all.sass'], // applies purging only on the bulma css file
+      },
+    }, // must be after other CSS plugins
+    'gatsby-plugin-netlify', // make sure to keep it last in the array
+  ],
+}
```
Comment (Shadowfiend) [inline gatsby-config.js:98]: Once we get tBTC and Keep to “I haz docs” land, I think we'll be in a good place to set up a template repo under Thesis that sets up the bare bones here.

File: package.json [hunk 2]
```diff
@@ -1,54 +1,64 @@
...
+    "gatsby-remark-images": "^3.1.6",
+    "gatsby-remark-relative-images": "^0.2.1",
+    "gatsby-source-filesystem": "^2.0.26",
+    "gatsby-transformer-remark": "^2.6.9",
+    "gatsby-transformer-sharp": "^2.1.9",
+    "lodash": "^4.17.15",
+    "lodash-webpack-plugin": "^0.11.4",
+    "netlify-cms-app": "^2.9.6",
+    "netlify-cms-media-library-cloudinary": "^1.3.2",
+    "netlify-cms-media-library-uploadcare": "^0.5.2",
```
Comment (Shadowfiend) [inline package.json:28]: Do we need these two media library plugins?

File: package.json [hunk 3]
```diff
@@ -1,54 +1,64 @@
...
+    "node-sass": "^4.11.0",
+    "pascal-case": "^3.1.1",
```
Comment (Shadowfiend) [inline package.json:30]: How.is.this.an.entire.package.

I'm 99.999% sure [this package](https://github.com/blakeembrey/change-case/tree/master/packages/pascal-case#readme) has more lines of configuration than actual code it contributes to the world.

< rage rage rage whine whine whine > (Also that package is published from a metapackage repo that has 21 packages including `is-lower-case` and `is-upper-case` so if we burn it all down let's start from right there.)

Incidentally I think we maybe [don't need it anymore](https://github.com/blakeembrey/change-case/tree/master/packages/pascal-case#readme).

**Comment by Shadowfiend** [comment]
> All less files are converted to scss files.

Oh the humanity!

File: package.json (hunk 3 above)
Comment (ironng) [reply package.json:30]: 😆 🙈 Yes, I'll 86 it.

File: package.json (hunk 2 above)
Comment (ironng) [reply package.json:28]: Mm...I'll look into it! TBH it's copy pasta 🍝 😬

File: gatsby-config.js (hunk 1 above)
Comment (ironng) [reply gatsby-config.js:98]: Sounds good!

File: package.json (hunk 2 above)
Comment (ironng) [reply package.json:28]: I'll take it out.

Comment (ironng) [reply package.json]: Removed!

**Comment by ironng** [comment]
Any opinions on 2 space vs 4 space indentation? I don't mind either way...

**Comment by Shadowfiend** [comment]
That sounds about right… Really need to hit all the sites with prettier.

**Comment by ironng** [comment]
Setting up prettier would be good! I’ll make a PR for exclusively for changing the spacing separately after this, to make reviewing changes easier.

**Comment by Shadowfiend** [comment]
I pushed the Actions stuff to `old-sport` so you could start seeing previews from this branch. `preview.keep.network` is hooked up, though `keep.network` still points to the current setup. We can switch it over in Cloudflare when the time comes!

**Comment by Shadowfiend** [comment]
Tbh my plan here is probably going to be to give the preview a once-over and then approve, since I don't think I'll have the cycles to deep review on the whole diff.

Let me know if there's anything that's known-broken so I can ignore it (“Redux/api calls will be hooked up in a separate PR” makes me think that might be the case)!

**Comment by ironng** [comment]
🙌 🙇‍♀️ Thank you for setting up the previews!!

Known broken stuff: email forms, animated scrolling to same page links, and highlighted states for same page nav links (i.e. highlighting “team” in the nav when you’re on the “team” section). The “see all” links might have the wrong margin on the right as well.

**Comment by Shadowfiend** [comment]
Whoops, had forgotten to add the GCP secret to publish this properly---fixed now, and we have previews.

**Comment by Shadowfiend** [comment]
Haven't done the accidental-close in a while, whee.

Let me know if you feel good in terms of merging this as-is, or if you want to add more.

**Comment by ironng** [comment]
Woo! Previews! 

Hmm...the preview build is not what I see locally. I’d prefer to fix the missing images before we merge.

**Comment by Shadowfiend** [comment]
Since the preview build does the prefixed build, any references that don't use Gatsby's prefixing will probably be broken---that's happening with links, maybe images too? Willing to wait for that to be fixed if you want though.

**Comment by ironng** [comment]
Aha. It’s the prefixing! I have do re-do how we handle the images anyhow...Gatsby has its own Img component that does optimizations

**Comment by Shadowfiend** [comment]
Ok merging here in that case :rocket:

**Pull request merged**